File: frontend/src/api/cosmeticEvents.ts
import { apiClient } from './client'

export type CosmeticProduct = {
  brand_name: string | null
  name_brand: string | null
  industry_code: string | null
  industry_name: string | null
}

export type CosmeticEventRecord = {
  report_number: string | null
  report_date: string | null
  serious: string | null
  outcomes: string[]
  reactions: string[]
  products: CosmeticProduct[]
}


export type CosmeticRecallNotice = {
  title: string | null
  product_name: string | null
  brand_name: string | null
  company_name: string | null
  category: string | null
  reason: string | null
  remedy?: string | null
  published_date: string | null
  record_url: string | null
  source_name: string
  source_kind?: 'structured_api' | 'public_notice' | 'normalized_public_notice'
  source_type?: string
  extraction_confidence?: string | null
}


export type CosmeticReaction = {
  reaction: string
  count: number
}

export type CosmeticSignalScore = {
  score: number
  label: string
  data_confidence: string
  top_reaction_concentration: number
  review_priority: string
  score_version: string
  limitations: string[]
}

export type CosmeticAudit = {
  audit_id: string
  source_id: string
  module: string
  upstream_status: string
  record_count: number
  transform_version: string
  source_snapshot_status: string | null
  source_pull_id: string | null
  source_payload_hash: string | null
}

export type CosmeticEventSearchResponse = {
  query: string
  raw_query?: string
  normalized_query?: string
  correction_applied?: boolean
  suggestion_message?: string | null
  count: number
  limit: number
  source_name: string
  endpoint: string
  retrieval_timestamp: string
  medical_disclaimer: string
  cosmetic_disclaimer: string
  audit: CosmeticAudit
  signal_score: CosmeticSignalScore
  top_reactions: CosmeticReaction[]
  records: CosmeticEventRecord[]
  recall_count?: number
  recall_source_name?: string | null
  recall_source_status?: string | null
  recall_source_error?: string | null
  recall_notices?: CosmeticRecallNotice[]
}

export async function searchCosmeticEvents(
  query: string,
  limit = 10
): Promise<CosmeticEventSearchResponse> {
  const response = await apiClient.get<CosmeticEventSearchResponse>(
    '/api/v1/cosmetic-events/search',
    {
      params: {
        q: query,
        limit,
      },
    }
  )

  return response.data
}

File: backend/app/schemas/cosmetic_events.py
from pydantic import BaseModel

from app.schemas.audit import AuditSummary


class CosmeticProduct(BaseModel):
    brand_name: str | None = None
    name_brand: str | None = None
    industry_code: str | None = None
    industry_name: str | None = None


class CosmeticEventRecord(BaseModel):
    report_number: str | None = None
    report_date: str | None = None
    serious: str | None = None
    outcomes: list[str]
    reactions: list[str]
    products: list[CosmeticProduct]



class CosmeticRecallNotice(BaseModel):
    title: str | None = None
    product_name: str | None = None
    brand_name: str | None = None
    company_name: str | None = None
    category: str | None = None
    reason: str | None = None
    published_date: str | None = None
    record_url: str | None = None
    source_name: str


class CosmeticReaction(BaseModel):
    reaction: str
    count: int


class CosmeticSignalScore(BaseModel):
    score: int
    label: str
    data_confidence: str
    top_reaction_concentration: float
    review_priority: str
    score_version: str
    limitations: list[str]


class CosmeticEventSearchResponse(BaseModel):
    query: str
    raw_query: str
    normalized_query: str
    correction_applied: bool
    suggestion_message: str | None = None
    count: int
    limit: int
    source_name: str
    endpoint: str
    retrieval_timestamp: str
    medical_disclaimer: str
    cosmetic_disclaimer: str
    audit: AuditSummary
    signal_score: CosmeticSignalScore
    top_reactions: list[CosmeticReaction]
    records: list[CosmeticEventRecord]
    recall_count: int = 0
    recall_source_name: str | None = None
    recall_source_status: str | None = None
    recall_source_error: str | None = None
    recall_notices: list[CosmeticRecallNotice] = []
